### PR TITLE
enable gateway support

### PR DESCRIPTION
**Pull request by xinlian12**
1. Add gateway support
2. Add connectionSharing support for direct mode

### CODE REVIEW

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 1]
```diff
@@ -55,6 +55,16 @@ public class CosmosDBConfig extends AbstractConfig {
...
 
+    public static final String COSMOS_GATEWAY_MODE_ENABLED = "connect.cosmos.connection.gateway.enabled";
+    private static final String COSMOS_GATEWAY_MODE_ENABLED_DOC =
+            "Flag to indicate whether to use gateway mode. By default it is false";
+    private static final boolean DEFAULT_COSMOS_GATEWAY_MODE_ENABLED = false;
+
+    public static final String COSMOS_CONNECTION_SHARING_ENABLED = "connect.cosmos.connection.sharing.enabled";
+    private static final String COSMOS_CONNECTION_SHARING_ENABLED_DOC =
+            "Flag to indicate whether connection sharing is enabled across different cosmosdb clients.";
+    private static final boolean DEFAULT_COSMOS_CONNECTION_SHARING_ENABLED = false;
```
Comment (FabianMeiswinkel) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:66]: Shouldn't this be true?

Comment (FabianMeiswinkel) [reply src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:66]: Never mind

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 2]
```diff
@@ -55,6 +55,16 @@ public class CosmosDBConfig extends AbstractConfig {
             "A comma delimited list of Kafka topics mapped to Cosmos containers.\n"
                     + "For example: topic1#con1,topic2#con2.";
 
+    public static final String COSMOS_GATEWAY_MODE_ENABLED = "connect.cosmos.connection.gateway.enabled";
+    private static final String COSMOS_GATEWAY_MODE_ENABLED_DOC =
+            "Flag to indicate whether to use gateway mode. By default it is false";
+    private static final boolean DEFAULT_COSMOS_GATEWAY_MODE_ENABLED = false;
+
+    public static final String COSMOS_CONNECTION_SHARING_ENABLED = "connect.cosmos.connection.sharing.enabled";
+    private static final String COSMOS_CONNECTION_SHARING_ENABLED_DOC =
```
Comment (kushagraThapar) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:64]: Worth mentioning more details about connection sharing. Also, usually connection sharing will happen for first client with other clients. So first client's connection settings will be used for other clients. 
If there is a way to set these up, we should mention these details in the doc.

**Review by kushagraThapar — APPROVED** [review]

File: src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java [hunk 3]
```diff
@@ -47,15 +47,21 @@ public void start(Map<String, String> map) {
...
+                        .endpoint(config.getConnEndpoint())
+                        .key(config.getConnKey())
+                        .connectionSharingAcrossClientsEnabled(this.config.isConnectionSharingEnabled())
+                        .userAgentSuffix(CosmosDBConfig.COSMOS_CLIENT_USER_AGENT_SUFFIX + version())
+                        .throttlingRetryOptions(
+                                new ThrottlingRetryOptions()
+                                        .setMaxRetryAttemptsOnThrottledRequests(Integer.MAX_VALUE)
+                                        .setMaxRetryWaitTime(Duration.ofSeconds((Integer.MAX_VALUE / 1000) - 1)));
+        if (this.config.isGatewayModeEnabled()) {
+            cosmosClientBuilder.gatewayMode();
```
Comment (FabianMeiswinkel) [inline src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java:61]: Do we need to change gateway mode defaults (number of connection in pool etc.)

Comment (xinlian12) [reply src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java:61]: The default pool size is 1000 by default, should be fine. keep the default as for now.

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java (hunk 2 above)
Comment (xinlian12) [reply src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:64]: how many cosmosdb clients will be created will be eventually decided by max.tasks, topic partitions etc - so I think it kind makes sense to use the same config for each one. Or if in the future we find the needs to have different config between clients we can then decide how to support it.

updated the doc to contain a little bit more info.

**Review by kushagraThapar — APPROVED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Pull request merged**